hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlebigquery/avro/GoogleBigQueryAvroToProtoConverter.java

```diff
@@ -14,65 +12,132 @@
+                messageBuilder.setField(field, toProtobufValue(field, fieldValue));
             }
-        });
+        }
+    }
+
+    private Integer toInteger(Object object) {
+        if (object instanceof Number) {
+            return ((Number) object).intValue();
+        } else if (object instanceof String) {
+            return Integer.parseInt((String) object);
+        } else if (object instanceof Boolean) {
+            return (Boolean) object ? 1 : 0;
+        } else {
+            return (int) object;
+        }
+
+    }
+
+    private Long toLong(Object object) {
+        if (object instanceof Number) {
+            return ((Number) object).longValue();
+        } else if (object instanceof String) {
+            return Long.parseLong((String) object);
+        } else if (object instanceof Boolean) {
+            return (Boolean) object ? 1l : 0l;
+        } else {
+            return (long) object;
+        }
 
-        return messageBuilder.build();
+    }
+
+
+    private <T extends Number> T toNumber(Object object, Class<T> clazz) {
+        if (object instanceof Number) {
+            return numberToSpecific((Number) object, clazz);
+        } else if (object instanceof String) {
+            return numberToSpecific(Double.parseDouble((String) object), clazz);
+        } else if (object instanceof Boolean) {
+            return numberToSpecific((Boolean) object ? 1 : 0, clazz);
+        } else return clazz.cast(object);
+
+    }
+
+    private <T extends Number> T numberToSpecific(Number object, Class<T> clazz) {
+        if (clazz == Integer.class) {
+            return clazz.cast((object).intValue());
+        } else if (clazz == Long.class) {
+            return clazz.cast((object).longValue());
+        } else if (clazz == Float.class) {
+            return clazz.cast((object).floatValue());
+        } else if (clazz == Double.class) {
+            return clazz.cast((object).doubleValue());
+        } else return clazz.cast(object);
     }
 
     private Object toProtobufValue(Descriptors.FieldDescriptor fieldDescriptor, Object value) {
         switch (fieldDescriptor.getType()) {
+            case UINT32:
             case INT32:
             case SINT32:
             case SFIXED32:
+                return toNumber(value, Integer.class);
             case INT64:
             case SINT64:
+            case FIXED64:
+            case UINT64:
             case SFIXED64:
-            case BOOL:
+                return toNumber(value, Long.class);
             case FLOAT:
+                return toNumber(value, Float.class);
             case DOUBLE:
-                return value;
+                return toNumber(value, Double.class);
             case STRING:
                 return value.toString();
+            case BOOL:
+                if (value instanceof Boolean) {
+                    return value;
+                } else if (value instanceof Number) {
+                    return ((Number) value).intValue() > 0;
```

Comment:
Is it really how the int field should be casted to boolean? I would have used `((Number) value).intValue() != 0`